packages/components/src/Layout/Semantics/Header.tsx

```diff
@@ -41,8 +42,11 @@ export const headerFooterCSS = css`
   flex: 0 0 auto;
 `
 
-export const Header = styled.header.withConfig({
-  shouldForwardProp,
-})<HeaderProps>`
+export const Header = styled.header
+  .withConfig({ shouldForwardProp })
+  .attrs<HeaderProps>(({ border }) => ({
+    border,
+  }))<HeaderProps>`
   ${headerFooterCSS}
+  ${({ border }) => border && addBorder(border)}
```

Comment:
`addBorder` helper should be able to pick-up `borderProps` without them being explicitly specified - see comment there.

```suggestion
  {addBorder}
```